### PR TITLE
added np.size() overload and added tests

### LINKED ISSUE #9464: The `np.size()` Function Fails
## The Bug

It seems Numba support the `ndarray` property of an array (See [`numpy.ndarray.size`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.size.html)) yet fails with the function [`np.size()`](https://numpy.org/doc/stable/reference/generated/numpy.ma.size.html).

## Code for Producing the Bug

The following works:

```python
import numpy as np
from numba import njit

@njit
def SumArrayWorks( vX : np.ndarray ) -> float:
  arrSum = 0.0
  for ii in range(vX.size):
    arrSum += vX[ii]

  return arrSum

vX = np.array([1.0, 2.0, 3.0, 4.0])
SumArrayWorks(vX) #<! Works!

```

While this will fail:

```python
import numpy as np
from numba import njit

@njit
def SumArrayFails( vX : np.ndarray ) -> float:
  arrSum = 0.0
  for ii in range(np.size(vX)):
    arrSum += vX[ii]

  return arrSum

vX = np.array([1.0, 2.0, 3.0, 4.0])
SumArrayFails(vX) #<! Fails!
```

The error is:

```python
TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Use of unsupported NumPy function 'numpy.size' or unsupported use of the function.
```

## System Information

1. Numba: `0.59.0`.
2. Python: `3.12`.
3. NumPy: `1.26.4`.
4. OS: Windows 11 Pro 23H2.

### PR DESCRIPTION
**Pull request by shourya5**
- [x] - resolves #9464
- [x] added unit tests for the np.size() function

### CODE REVIEW

**Comment by kc611** [comment]
Hi @shourya5, Thank you for your contribution. :smile: 

As this feature will be mentioned in our next release notes, could you add a short description of this feature as a small release note `.rst` file. [Here are the instructions to do so.](https://github.com/numba/numba/blob/main/docs/upcoming_changes/README.rst)

File: numba/tests/test_array_manipulation.py [hunk 1]
```diff
@@ -174,6 +174,9 @@ def numpy_fill_diagonal(arr, val, wrap=False):
 def numpy_shape(arr):
     return np.shape(arr)
 
+def numpy_size(arr):
```
Comment (kc611) [inline numba/tests/test_array_manipulation.py:177]: This function also needs a corresponding runner function. See: https://github.com/numba/numba/blob/108560408eb3456c53bb08426eec1ea62f95ff12/numba/tests/test_array_manipulation.py#L1198

File: numba/tests/test_record_dtype.py [hunk 2]
```diff
@@ -332,6 +332,10 @@ def get_shape(rec):
     return np.shape(rec.j)
 
 
+def get_size(rec):
+    return np.size(rec.j)
```
Comment (kc611) [inline numba/tests/test_record_dtype.py:336]: Same as above stated comment. This needs a corresponding runner function.

**Review by kc611 — CHANGES_REQUESTED** [review]

File: numba/tests/test_array_manipulation.py (hunk 1 above)
Comment (shourya5) [reply numba/tests/test_array_manipulation.py:177]: > This function also needs a corresponding runner function. See:
> 
> https://github.com/numba/numba/blob/108560408eb3456c53bb08426eec1ea62f95ff12/numba/tests/test_array_manipulation.py#L1216

If I implement this loop in test_size() it fails for some reason

Comment (shourya5) [reply numba/tests/test_array_manipulation.py:177]: > This function also needs a corresponding runner function. See:
> 
> https://github.com/numba/numba/blob/108560408eb3456c53bb08426eec1ea62f95ff12/numba/tests/test_array_manipulation.py#L1216

I can't seem to replicate this loop pattern in test_size() without the tests failing,any tips?

File: docs/upcoming_changes/9504.np_support.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+added support for ``np.size()``
+====================================================
+
+added ``np.size()`` support for numpy,which was previously unsupported.
```
Comment (guilhermeleobas) [inline docs/upcoming_changes/9504.np_support.rst:4]:
```suggestion
added ``np.size()`` support for numpy, which was previously unsupported.
```

File: numba/tests/test_array_manipulation.py [hunk 4]
```diff
@@ -1218,6 +1221,31 @@ def check(x):
 
         self.assertIn("The argument to np.shape must be array-like",
                       str(raises.exception))
+        
+    def test_size(self):
+        pyfunc = numpy_size
+        cfunc = jit(nopython=True)(pyfunc)
```
Comment (guilhermeleobas) [inline numba/tests/test_array_manipulation.py:1227]:
```suggestion

    def test_size(self):
        pyfunc = numpy_size
        cfunc = jit(nopython=True)(pyfunc)
```

File: docs/upcoming_changes/9504.np_support.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+added support for ``np.size()``
+====================================================
+
+added ``np.size()`` support for numpy, which was previously unsupported.
```
Comment (guilhermeleobas) [inline docs/upcoming_changes/9504.np_support.rst:4]:
```suggestion
Added support for ``np.size()``
-------------------------------

Added ``np.size()`` support for numpy, which was previously unsupported.
```

**Review by guilhermeleobas — APPROVED** [review]
Thanks @shourya5. Just waiting for @kc611 approval and this one can be merged.

**Review by kc611 — APPROVED** [review]
LGTM!

Thank you @shourya5 @guilhermeleobas

**Pull request merged**